shared/src/main/scala/ListHelpers.scala

```diff
@@ -67,6 +99,55 @@ object ListHelpers:
     moldHelper(content, shape, 0)
   end mold
 
+  def sortBy(iterable: VList, key: VFun)(using ctx: Context): VList =
+    val branches = key.originalAST match
+      case Some(lam) => lam.body
+      case None =>
+        return VList(
+          iterable.zipWithIndex
+            .sorted((a, b) =>
+              MiscHelpers.compareExact(
+                key.executeResult(a(0), a(1), List(a(0))),
+                key.executeResult(b(0), b(1), List(b(0)))
+              )
+            )
+            .map(_._1)*
+        )
+
+    if branches.length < 2 then
+      return VList(
+        iterable.zipWithIndex
+          .sorted((a, b) =>
+            MiscHelpers.compareExact(
+              key.executeResult(a(0), a(1), List(a(0))),
+              key.executeResult(b(0), b(1), List(b(0)))
+            )
+          )
+          .map(_._1)*
+      )
+
+    val out = iterable.zipWithIndex
+      .sortWith { (a, b) =>
+        val (aRes, bRes) =
+          branches
+            .view
+            .map { branch =>
+              val f = VFun.fromLambda(AST.Lambda(1, List.empty, List(branch)))
+              (
+                f.execute(a(0), a(1), List(a(0))),
+                f.execute(b(0), b(1), List(b(0)))
+              )
+            }
+            .dropWhile(_ == _)
+            .head
```

Comment:
```suggestion
            .find(_ != _)
            .getOrElse(ctx.settings.defaultValue)
```

Reply:
What does this do?

Reply:
If they compare equal for all branches, `.dropWhile(_ == _)` would return an empty list, so `.head` would have thrown an exception. This one finds the first branch where they're not equal, and if such a branch can't be found, gives `(0, 0)` (or whatever the default value is set to be)